app/main.py

```diff
@@ -1,2 +1,86 @@
+from typing import Any, Iterable, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8, load_factor: float = 0.67) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.threshold = int(self.capacity * self.load_factor)
+        self.length = 0
+        self.hash_table = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = hash(key) % len(self.hash_table)
+        while self.hash_table[index] and self.hash_table[index][0] != key:
+            index = (index + 1) % len(self.hash_table)
+        if self.hash_table[index] and self.hash_table[index][0] == key:
+            self.hash_table[index] = (key, hash(key), value)
+        else:
+            self.hash_table[index] = (key, hash(key), value)
```

Comment:
no need to use every time hash() function, use it before the loop and write it into variable